src/main/java/org/gbif/ipt/action/portal/ResourceFileAction.java

```diff
@@ -257,4 +258,33 @@ public String rtf() {
     filename = sb.toString();
     return execute();
   }
+
+  public String rawsource() {
+    if (resource == null) {
```

Comment:
I think we can have only one if
`if (resource == null || id == null)`

Reply:
Done, but in the other PR (bootstrap): https://github.com/gbif/ipt/pull/1546